odoo/addons/base/models/ir_qweb.py

```diff
@@ -861,6 +862,31 @@ def _load(self, ref):
 
     # values for running time
 
+    def _get_converted_image_data_uri(self, base64_source):
+        if self.env.context.get('pdf'):
+            mimetype = FILETYPE_BASE64_MAGICWORD.get(base64_source[:1], 'png')
+            if 'webp' in mimetype:
+                # Use converted image so that is recognized by wkhtmltopdf.
+                bin_source = base64.b64decode(base64_source)
+                Attachment = self.env['ir.attachment']
+                checksum = Attachment._compute_checksum(bin_source)
+                origin = Attachment.sudo().search([
+                    ['id', '!=', False],  # No implicit condition on res_field.
+                    ['checksum', '=', checksum],
+                ])
+                if origin:
+                    origin_ids = [attachment.id for attachment in origin]
+                    converted_domain = [
+                        ['id', '!=', False],  # No implicit condition on res_field.
+                        ['res_model', '=', 'ir.attachment'],
+                        ['res_id', 'in', origin_ids],
+                        ['mimetype', '=', 'image/jpeg'],
```

Comment:
It is to always expect the same flow, so you could say it is for simplicity indeed.

1) On upload in website: suggest conversion to webp
2) On webp upload:
- 2A) upload converted jpg
- 2B) upload resized webp (and therefore jpg of each resize because of (2A))

There is currently no "fallback to original" in (2A).

Reply:
Crystal clear, would you mention that in the PR message?
I think this one is going to be a rebase-merge, we will see, might be a good thing to mention it in a commit message if not the case already (I don't think it is?).

Reply:
:point_up: